PrimerKlarnaSDK/Classes/PrimerKlarnaViewController.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  PrimerKlarnaViewController.swift
+//  PrimerKlarnaSDK
+//
+//  Created by Evangelos on 22/8/22.
+//
+
+#if canImport(UIKit)
+
+import KlarnaMobileSDK
+import UIKit
+
+public enum KlarnaPaymentCategory: String {
+    case payNow = "pay_now", payLater = "pay_later", payOverTime = "pay_over_time"
+}
+
+public protocol PrimerKlarnaViewControllerDelegate {
+    func primerKlarnaViewDidLoad()
+    func primerKlarnaPaymentSessionCompleted(authorizationToken: String?, error: PrimerKlarnaError?)
+}
+
+public class PrimerKlarnaViewController: UIViewController {
+    
+    var klarnaPaymentView: KlarnaPaymentView!
+    var delegate: PrimerKlarnaViewControllerDelegate
+    private var paymentCategory: KlarnaPaymentCategory
+    private var clientToken: String
+    private var urlScheme: String?
+    private var klarnaPaymentViewHeightConstraint: NSLayoutConstraint!
+    
+    public init(
+        delegate: PrimerKlarnaViewControllerDelegate,
+        paymentCategory: KlarnaPaymentCategory,
+        clientToken: String,
+        urlScheme: String?)
+    {
+        self.delegate = delegate
+        self.paymentCategory = paymentCategory
+        self.clientToken = clientToken
+        self.urlScheme = urlScheme
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .white
+        
+        klarnaPaymentView = KlarnaPaymentView(category: self.paymentCategory.rawValue, eventListener: self)
+        view.addSubview(klarnaPaymentView)
+        view.heightAnchor.constraint(equalToConstant: 800).isActive = true
```

Comment:
Not sure if needed, but we should double-check how this would behave on smaller devices.